_posts/2016-05-24-SpringIO16-Spring-Rest-Docs.md

```diff
@@ -0,0 +1,339 @@
+                                fieldWithPath("linkedin").description("The employee's linkedin"),
+                                fieldWithPath("unit").description("The employee's unit").type(Unit.class),
+                                fieldWithPath("_links").description("links to other resources")
+                        ));
+ mockMvc.perform(
+                get("/employees/1").accept(MediaType.APPLICATION_JSON));
+
+{% endhighlight %}
+
+
+#### Relaxed Snippet (New in 1.1)
+In contrast to default snippets, relaxed snippets are used to have no complaining when something was neglected in the document.
+This has the advantage if you would like to focus on a certain scenario or specific part of the response/request.
+
+#### Reusable Snippet (New in 1.1)
+With the newly introduced reusable snippet, you can define a snippet in the beginning of your test class and reuse it every time you need it. 
+When added to your test method, you can extend it with extra variables. 
+
+
+{% highlight java %}
+
+// First we define a snippet for reuse
+protected final LinksSnippet pagingLinks = links(
+        linkWithRel("first").optional().description("The first page of results"),
+        linkWithRel("last").optional().description("The last page of results"),
+        linkWithRel("next").optional().description("The next page of results"),
+        linkWithRel("prev").optional().description("The previous page of results"));
+
+// Then you perform the mockMvc and add the snippet to the document.
+// As you can see, it is expendable.
+this.mockMvc.perform(get("/").accept(MediaType.APPLICATION_JSON))
+    .andExpect(status().isOk())
+    .andDo(document("example", this.pagingLinks.and( 
+            linkWithRel("alpha").description("Link to the alpha resource"),
+            linkWithRel("bravo").description("Link to the bravo resource"))));
+{% endhighlight %}
+
+
+## Type of Snippets:
+A snippet can be one of the following:
+
+### Hypermedia links
+When documenting your hypermedia application, you'll have to define your links and what they link to. 
+If you have dynamic links that can disappear at one time, you can use relaxed snippets or optional so it won't complain.
+
+{% highlight java %}
+ this.document.snippets(
+                links(
+                        halLinks(), linkWithRel("self").description("The employee's resource"),
+                        linkWithRel("employee").optional().description("The employee's projection")),
+                responseFields(
+                        fieldWithPath("username").description("The employee unique database identifier").type(String.class),
+                        fieldWithPath("_links").description("links to other resources")
+                ));
+{% endhighlight %}
+
+
+### Request fields
+This defines your fields you request from the client.
+Normally it will complain if you neglected a field but with v1.1 Spring REST Docs has support for Relaxed Snippets.
```

Comment:
.... complain when you neglect a field but since v1.1 Spring REST Docs supports Relaxed Snippets